x/mongo/driver/topology/connection_test.go

```diff
@@ -958,3 +958,40 @@ func (t *testCancellationListener) assertMethodsCalled(testingT *testing.T, numL
 	assert.Equal(testingT, numStopListening, t.numStopListening, "expected StopListening to be called %d times, got %d",
 		numListen, t.numListen)
 }
+
+var testCases = []struct {
+	name   string
+	buffer []byte
+	desc   description.Server
+}{
+	{
+		"message too large errors with small max message size",
+		[]byte{0x0A, 0x00, 0x00, 0x00, 0x05, 0x06, 0x07, 0x08, 0x09, 0x0A},
+		description.Server{MaxMessageSize: 9},
+	},
+	{
+		"message too large errors with default max message size",
+		append([]byte{0x04, 0x6C, 0xDC, 0x02}, make([]byte, 4800000)...),
+		description.Server{},
+	},
+}
+
+func TestReadWireMessageTooLarge(t *testing.T) {
```

Comment:
This should go as a sub-test of the larger-scoped `TestConnection` as it was before and the test cases should be declared as a local variable inside the test. See https://github.com/mongodb/mongo-go-driver/blob/7841aba303d04e74c006bcd7d6fbed4fee4b87a2/mongo/integration/collection_test.go#L223 for an example.

Reply:
Cool, sounds good. Moved back to a sub-test of `TestConnection`.